ramp-engine/ramp_engine/dispatcher.py

```diff
@@ -174,14 +178,23 @@ def collect_result(self, session):
             elif self.hunger_policy == 'exit':
                 self._poison_pill = True
             return
+
         for worker, (submission_id, submission_name) in zip(workers,
                                                             submissions):
+            dt = worker.time_since_last_status_check()
+            wait_longer = (False if dt is None else
+                           dt < self.status_check_wait)
+            if wait_longer:
+                self._processing_worker_queue.put_nowait(
+                    (worker, (submission_id, submission_name)))
+                time.sleep(0)
+                continue
             if worker.status == 'running':
```

Comment:
I think that we can safely put an `elif` here

```suggestion
            elif worker.status == 'running':
```